### ISSUE TITLE
Injecting dependencies into Middleware (constructor) causes no 'resolve' method (v6)

### ISSUE THREAD

**Issue by appsolutegeek** [nestjs/nest#1834]
<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
According to the documentation, dependencies can be injected directly into the constructor of Middleware.

I got the following error

```
[2019-03-25T10:37:55.536] [FATAL] Bootstrap - An error as occured in bootstrapping the application { Error: The middleware doesn't provide the 'resolve' method (CorrelationMiddleware)
    at MiddlewareModule.bindHandler
```

The resolve seems to have been deprecated, so I am unsure why its complaining.

Currently, my correlationSerivce has only getters and setter, nothing complex.

If I remove the middleware constructor the app compiles, and I can also inject the correlationService into a controller - for example, and everything works - so I know my CorrelationService provider is working.

```
@Injectable()
export class CorrelationMiddleware implements NestMiddleware {
  constructor(private correlationService: CorrelationService) {
    console.log("in constructor on the corrrelationMiddleware")
  }

  use(request: Request, response: Response, next: Function) {
    console.log("INSIDE THE MIDDLEWARE!!!")
    const correlationId = v4()
    request.correlationId = correlationId
    //this.correlationService.setCorrelationId(correlationId)
    next()
  }
}
```

and the the module.

```
@Module({
  providers: [CorrelationService],
  exports: [CorrelationService]
})
export class CorrelationModule implements NestModule {
  configure(consumer: MiddlewareConsumer): void | MiddlewareConsumer {
    consumer.apply(CorrelationMiddleware).forRoutes("*")
  }
}
```

## Expected behavior
 
As per documentation, you should be able to inject providers into middleware.

## Environment

<pre><code>
Nest version: 6
 
For Tooling issues:
- Node version: 11
- Platform:  MAc

</code></pre>

**Comment by kamilmysliwiec** [comment]
> The middleware doesn't provide the 'resolve' method (CorrelationMiddleware)
    at MiddlewareModule.bindHandler

It looks like an issue cause by inconsistent packages (v5 used together with v6). Please, update all `@nestjs` packages to the latest versions.

**Comment by appsolutegeek** [comment]
Ermrm, I just went through alll my project and remove teh packge lock and removed node_modules and reinstalled it all. I am going to buld a simple project with the nest cli and test it ...  

I will come back and confirm and if its still the same i will upload a github repo

**Comment by appsolutegeek** [comment]
OK - I found this issue, I am going to upload a repo... but basically its because the provider is marked as {scope: Scope.Request} - if I comment it out - then it works...

https://github.com/appsolutegeek/middleware-error-nest

I need a new instance of the provider each request, its for adding a correlationId.

The scope.Request works when injecting into controller / other provider.

**Comment by appsolutegeek** [comment]
The above project was created with the nest cli, here is the error that is output.

(node:53750) UnhandledPromiseRejectionWarning: Error: The middleware doesn't provide the 'resolve' method (CorrelationMiddleware)

**Comment by appsolutegeek** [comment]
I have a workaround but I would like to know if this is possible above?

You fixed the docs for the Scope.Request because of my other submission and now I notice I can inject the request object into a scoped provider .. so my middleware I add the correlationID to the request and then in the provider I am able to get the request object :-)

But it would be nice to know if its possible with middleware, right now, it throws the above error about the  resolve method.

**Comment by kamilmysliwiec** [comment]
Thanks for reporting! Fixed in 6.0.4.
